explainer.md

```diff
@@ -57,6 +57,25 @@ This explainer does not address Anchors (see [anchors](https://github.com/immers
+
+The following diagram should help in understanding the underlying issue:
+<img alt="Hit-test example" src="explainer-media/timing-diagram.png" style="float: left; object-fit: cover; width: 45%; height: 20em; margin-right: 1em; ">
+
+Situation: call requestHitTest(input ray) at frame N. 
+Problem: depending on the use-case, location 1 or location 2 may be the desired location of the result. Because the surface hit may be several meters away, small device rotations can result in large discrepancies between locations 1 and 2.
+
+Use-Cases:
+*   Reticle
+    *   For a pixel-perfect reticle, location 2 is the correct result. We need the reticle to be rendered in frame N+1 at location 2, otherwise it will appear to lag behind the device.
+    *   This means we either need to use a 2d API for reticles since 2d remains accurate from frame to frame (is there a way this could work for HMDs?), or we need to automatically correct the 3d input ray to the current device ray by adding the delta between the device transform at frame N and N+1.
+*   Object placement
+    *   For accurate object placement, it's actually correct to use location 1. If the placement is based on an input event / touch event, you want the object to appear in the location that it was at - for example when it was being displayed as a preview on the reticle - when the event was triggered. This means you DON'T want location 2 as that will cause the object to appear to have jumped from where you tried to place it.
+
+It is very important to note that location 1 and 2 are both valid depending on the context of the hit request. We could take steps to ensure that either location 1 or 2 would be returned but neither is correct in all cases. Furthermore, in the code where we are computing the result, we lack the crucial context information to understand which of the cases we are dealing with.
+
+In order to support these use-cases, a relative ray and a frame-of-reference acquired for a device or input-source can be supplied to the hit-test API. The frame-of-reference for devices and input-sources will be automatically and dynamically updated when a new frame is generated and prior to the hit-test logic. An absolute ray will be calculated based on the relative ray and the updated frame-of-reference for the device and then the hit-test results will be returned prior to the callback for the frame. This lets the API support both object placement with an absolute ray relative to a device or an absolute ray relative to an "eye-level" WebXR coordinate system.
```

Comment:
So to clarify, if I use a session's general FOR ("eye-level"), it'll resolve in the next frame in object-placement mode, but if I use the device's FOR, it'll be in reticle mode? Is this correct?

```js
async function () {
  const frameOfRef = await session.requestFrameOfReference('eye-level');
  const location1 = await session.requestHitTest(this.camera, [0,0,1], frameOfRef);

  const location2 = await session.requestHitTest(this.camera, [0, 0, 1], xrDevice); // however you generate the FOR from device
}
```

Also I guess `origin` is no longer necessary for the reticle-mode version. Looks good, but could be more explicit on the semantics here (maybe with some sketch code)

Reply:
I would prefer to discuss in terms of "absolute vs. relative" as opposed to object-placement vs. reticle - there aren't really modes per-se. That said, I believe you are correct in your assessment. I can't think of a reason to use any origin other than (0, 0, 0) when doing a test relative to a device FOR, but I think it's overall much simpler to not have two different APIs.